src/plugins/discover/public/application/apps/not_found/not_found_route.tsx

```diff
@@ -34,14 +36,21 @@ export function NotFoundRoute(props: NotFoundRouteProps) {
       defaultMessage: 'Page not found',
     });
 
-    bannerId = services.core.overlays.banners.replace(
+    bannerId = core.overlays.banners.replace(
       bannerId,
       toMountPoint(
         <EuiCallOut color="warning" iconType="iInCircle" title={bannerMessage}>
           <p>
             <FormattedMessage
               id="discover.noMatchRoute.bannerText"
-              defaultMessage="Invalid URL for Discover application."
+              defaultMessage="Discover application doesn't recognize this route: {route}"
+              values={{
+                route: (
```

Comment:
I don't know if we need this tbh, seems like a bit of an overkill. Also, I would strongly suggest removing referring back to the same route, I see no point in this.

Reply:
I have added it, since the same approach was applied to [visualize](https://github.com/elastic/kibana/blob/7162f16f7a4d55be767d9c007cfb370e8d7709aa/src/plugins/visualize/public/application/components/visualize_no_match.tsx#L44) and [dashboard](https://github.com/elastic/kibana/blob/7162f16f7a4d55be767d9c007cfb370e8d7709aa/src/plugins/dashboard/public/application/listing/dashboard_no_match.tsx#L44). 
Do not have objections to remove it and have smth like "`nvalid URL for Discover application was requested.`. What do you think @timroes?

Reply:
I would share your idea about keeping the approaches aligned with Dashboard and Visualize. Also I think there might be a benefit of easier debuggability if anyone runs into this if we see the route. That said I'd agree with Maja, that we should not link to that route (which actually doesn't even happen here, since we just link back to the current page - the start page - again :D). So I'd keep the message, but remove the link.